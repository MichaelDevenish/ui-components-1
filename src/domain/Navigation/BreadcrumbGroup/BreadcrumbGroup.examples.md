#### BreadcrumbGroup

```jsx
<BreadcrumbGroup>
  <BreadcrumbGroup.Breadcrumb href='/test'>
    Dashboard
  </BreadcrumbGroup.Breadcrumb>
   <BreadcrumbGroup.Breadcrumb href='/test' anchorComponentProps={{useReactRouter: false}}>
    Tasks
  </BreadcrumbGroup.Breadcrumb>
   <BreadcrumbGroup.Breadcrumb href='/test' type='active'>
    Create
  </BreadcrumbGroup.Breadcrumb>
</BreadcrumbGroup>
```

#### Legacy Support

```jsx
<div class="ihr-breadcrumbs">
    <a class="ihr-breadcrumbs__crumb">Dashboard</a>
    <a class="ihr-breadcrumbs__crumb">Tasks</a>
    <a class="ihr-breadcrumbs__crumb--active">Create</a>
</div>
```
