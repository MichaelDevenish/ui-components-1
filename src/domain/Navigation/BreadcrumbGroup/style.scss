@import '../../../common/sass/variables';

@mixin base-crumb {
  color: $i400;
  float: left;
  font-size: 12px;

  a {
    color: $i400;

    &:hover {
      text-decoration: underline;
    }
  }

  &:not(:last-child) {
    &::after {
      color: #667380;
      content: '/';
      margin: 0 12px;
      opacity: 1;
      position: relative;
      top: 0;
    }
  }
}

:local(.breadcrumbs),
.ihr-breadcrumbs {
  list-style: none;
  margin: 0;

  &::before,
  &::after {
    content: ' ';
    display: table;
  }
}

:local(.baseCrumb),
.ihr-breadcrumbs__crumb {
  @include base-crumb;
}

:local(.activeCrumb),
.ihr-breadcrumbs__crumb--active {
  @include base-crumb;

  &,
  a {
    color: $n700;
  }
}
