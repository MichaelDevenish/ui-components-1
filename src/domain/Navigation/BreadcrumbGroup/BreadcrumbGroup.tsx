import React from 'react'

import { Props } from '../../../common'
import { Breadcrumb } from './subcomponents/Breadcrumb'

import style from './style.scss'

interface IBreadcrumbGroupProps {
  /** The data-component-context */
  componentContext?: string
}

class BreadcrumbGroup extends React.Component<IBreadcrumbGroupProps> {
  public static Breadcrumb = Breadcrumb

  public render (): JSX.Element | null {
    const {
      children,
      componentContext
    } = this.props

    return (
      <div
        className={style.breadcrumbs}
        data-component-type={Props.ComponentType.BreadcrumbGroup}
        data-component-context={componentContext}
      >
        {children}
      </div>
    )
  }
}

export {
  BreadcrumbGroup
}
