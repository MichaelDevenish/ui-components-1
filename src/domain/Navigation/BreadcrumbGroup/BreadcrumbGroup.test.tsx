import { mount } from 'enzyme'
import React from 'react'

import { BreadcrumbGroup } from './BreadcrumbGroup'

describe('<PageLayout />', () => {
  it(`should render a simple breadcrumbGroup`, () => {
    const wrapper = mount(
      <BreadcrumbGroup>
        <BreadcrumbGroup.Breadcrumb to='/test'>
          Dashboard
        </BreadcrumbGroup.Breadcrumb>
        <BreadcrumbGroup.Breadcrumb to='/test' useReactRouter={false}>
          Tasks
        </BreadcrumbGroup.Breadcrumb>
        <BreadcrumbGroup.Breadcrumb to='/test' type='active'>
          Create
        </BreadcrumbGroup.Breadcrumb>
      </BreadcrumbGroup>
    )

    expect(wrapper).toMatchSnapshot()
  })

  it(`should render a breadcrumb by itself`, () => {
    const wrapper = mount(
      <BreadcrumbGroup.Breadcrumb to='/test'>
        Create
      </BreadcrumbGroup.Breadcrumb>
    )

    expect(wrapper).toMatchSnapshot()
  })
})
