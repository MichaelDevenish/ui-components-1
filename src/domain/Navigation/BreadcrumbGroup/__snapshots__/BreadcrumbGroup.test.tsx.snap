// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BreadcrumbGroup /> should render a breadcrumb by itself 1`] = `
<Breadcrumb
  anchorComponentProps={
    Object {
      "useReactRouter": true,
    }
  }
  href="/test"
  type="base"
>
  <Anchor
    anchorComponentProps={
      Object {
        "useReactRouter": true,
      }
    }
    className="baseCrumb"
    data-component-type="breadcrumb"
    href="/test"
    openInNewTab={false}
  >
    <a
      className="baseCrumb"
      data-component-type="breadcrumb"
      href="/test"
    >
      Create
    </a>
  </Anchor>
</Breadcrumb>
`;

exports[`<BreadcrumbGroup /> should render a simple breadcrumbGroup 1`] = `
<BreadcrumbGroup>
  <div
    className="breadcrumbs"
    data-component-type="breadcrumb_group"
  >
    <Breadcrumb
      anchorComponentProps={
        Object {
          "useReactRouter": true,
        }
      }
      href="/test"
      type="base"
    >
      <Anchor
        anchorComponentProps={
          Object {
            "useReactRouter": true,
          }
        }
        className="baseCrumb"
        data-component-type="breadcrumb"
        href="/test"
        openInNewTab={false}
      >
        <a
          className="baseCrumb"
          data-component-type="breadcrumb"
          href="/test"
        >
          Dashboard
        </a>
      </Anchor>
    </Breadcrumb>
    <Breadcrumb
      anchorComponentProps={
        Object {
          "useReactRouter": false,
        }
      }
      href="/test"
      type="base"
    >
      <Anchor
        anchorComponentProps={
          Object {
            "useReactRouter": false,
          }
        }
        className="baseCrumb"
        data-component-type="breadcrumb"
        href="/test"
        openInNewTab={false}
      >
        <a
          className="baseCrumb"
          data-component-type="breadcrumb"
          href="/test"
        >
          Tasks
        </a>
      </Anchor>
    </Breadcrumb>
    <Breadcrumb
      anchorComponentProps={
        Object {
          "useReactRouter": true,
        }
      }
      href="/test"
      type="active"
    >
      <Anchor
        anchorComponentProps={
          Object {
            "useReactRouter": true,
          }
        }
        className="activeCrumb"
        data-component-type="breadcrumb"
        href="/test"
        openInNewTab={false}
      >
        <a
          className="activeCrumb"
          data-component-type="breadcrumb"
          href="/test"
        >
          Create
        </a>
      </Anchor>
    </Breadcrumb>
  </div>
</BreadcrumbGroup>
`;
